sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -191,6 +151,47 @@ public void fail(@Nonnull Throwable t) {
         return result;
     }
 
+    public CompletableFuture<Long> executeAlterTableRenameTable(AlterTableRenameAnalyzedStatement statement) {
+        DocTableInfo sourceTableInfo = statement.sourceTableInfo();
+        TableIdent sourceTableIdent = sourceTableInfo.ident();
+        TableIdent targetTableIdent = statement.targetTableIdent();
+
+        return closeTable(sourceTableInfo)
+            .thenCompose(r -> renameTable(sourceTableIdent, targetTableIdent))
+                .whenComplete((r, f) -> {
+                    TableIdent tableToOpen = targetTableIdent;
```

Comment:
I'll add a FIXME as the related information if a table is closed will be introduced by another PR